src/reader.rs

```diff
@@ -95,18 +97,17 @@ impl<R: Read> Inner<R> {
             let before_out = self.stream.total_out();
             let rc = f(&mut self.stream, &self.buf[self.pos..self.cap], eof);
             self.pos += (self.stream.total_in() - before_in) as usize;
-            let read = (self.stream.total_out() - before_out) as usize;
+            read += (self.stream.total_out() - before_out) as usize;
```

Comment:
Note: this _does_ change the return value if we go through this loop multiple times.  This return value makes more sense to me: that being the size read, starting from the beginning of the method call - rather than the size read _in the last iteration_.  The latter doesn't make much sense to me.